### ISSUE TITLE
[Bug] FullScreenMedia range error

### ISSUE THREAD

**Issue by EtcGonza** [GetStream/stream-chat-flutter#1097]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

 stream_chat_flutter: ^3.4.0

### What happened?

Hi! I found a bug in the FullScreenMedia and GalleryHeader widgets.

In my app I have a copy of the **ChannelMediaDisplayScreen** of the official stream_chat_v1 sample. [This is the direct link of the samples](https://github.com/GetStream/flutter-samples/tree/main/packages/stream_chat_v1)

FullScreenMedia recive a list of attachments so you can swipe between those if you have more than one also this widget have a reference of the current page of the attachment that you are seeing. 
So if you pass five attachments to the widget and swipe until the second attachment the valueof the current page is one.

This is my code of the FullScreenMedia

```
InkWell(
                  onTap: () {
                    Navigator.push(
                      context,
                      MaterialPageRoute(
                        builder: (context) => StreamChannel(
                          channel: channel,
                          child: FullScreenMedia(
                            startIndex: position,
                            mediaAttachments:
                                media.map((e) => e.attachment).toList(),
                            message: media[position].message,
                            userName: media[position].message.user!.name,
                            // at.copyWith(showReply: false),
                          ),
                        ),
                      ),
                    );
                  },
                  child: SomeWidget
```

Inside FullScreenMedia there is a GalleryHeader widget. This widget recieve a message and a currentIndex and the currentIndex that is passing right now is the value of currentPage that have FullScreenMedia. And this is wrong because GalleryHeader have a function called **_showMessageActionModalBottomSheet** and its trying to access to the attachment list of the message with this value. This cause a exception because the currentPage could be five and the message attachment could be one.

This is the code of the function _showMessageActionModalBottomSheet in the GalleryHeader

```
  void _showMessageActionModalBottomSheet(BuildContext context) async {
   ///some code

    final defaultModal = AttachmentActionsModal(
      message: message,
      currentIndex: currentIndex,
      onShowMessage: onShowMessage,
    );

    final effectiveModal = attachmentActionsModalBuilder?.call(
          context,
          message.attachments[currentIndex],
          defaultModal,
        ) ??
        defaultModal;

    ///some code
```

The index that recives GalleryHeader It must not be the same as the one in the gallery. Because the gallery have the index of the attachment page and the gallery use this index to access to the message's attachment. This cause an exception.

There is no error if the FullScreenMedia recives a list of one attachment this error happens when there is more than one attachments.

### Relevant log output

```shell
This is the exception that I get when I tap on the three dots button in the FullScreenMedia widget.
I/flutter ( 9304): RangeError (index): Invalid value: Only valid value is 0: 1
```

- [X] I agree to follow this project's Code of Conduct

**Comment by imtoori** [comment]
Thanks for reporting the issue, but it has already been fixed in v4 https://github.com/GetStream/stream-chat-flutter/issues/1071

**Comment by EtcGonza** [comment]
> Thanks for reporting the issue, but it has already been fixed in v4 #1071

Oh great! There is release date for this v4?

**Comment by EtcGonza** [comment]
> there's a beta already out! the stable one will be released by the end of April

Thanks for the answer! I wait for v4.
